src/Application/TodoItems/Commands/CreateTodoItem/CreateTodoItemCommand.cs

```diff
@@ -25,16 +25,11 @@ public CreateTodoItemCommandHandler(IApplicationDbContext context)
 
         public async Task<int> Handle(CreateTodoItemCommand request, CancellationToken cancellationToken)
         {
-            var entity = new TodoItem
-            {
-                ListId = request.ListId,
-                Title = request.Title,
-                Done = false
-            };
+            var entity = new TodoItem(request.ListId, request.Title);
 
             entity.DomainEvents.Add(new TodoItemCreatedEvent(entity));
 
-            _context.TodoItems.Add(entity);
+            await _context.TodoItems.AddAsync(entity, cancellationToken);
```

Comment:
The AddAsync method should be used only in certain cases, you can read about it on the Microsoft website